core/src/main/java/org/infinispan/stream/impl/DistributedCacheStream.java

```diff
@@ -789,6 +789,12 @@ public void forEachOrdered(Consumer<? super R> action) {
       return this;
    }
 
+   @Override
+   public CacheStream<R> filterKeySegments(IntSet segments) {
+      segmentsToFilter = segments;
+      return this;
```

Comment:
Can't we do this in `AbstractCacheStream`?

Reply:
Unfortunately, no. The generics are kinda wonky on this due to `CacheStream` extending `Stream` and this guy implementing `BaseStream`.